util/json.go

```diff
@@ -77,12 +77,55 @@ func MustMarshalJSON(x interface{}) []byte {
 // Thereby, it is converting its argument to the representation expected by
 // rego.Input and inmem's Write operations. Works with both references and
 // values.
-func RoundTrip(x *interface{}) error {
+func RoundTrip(x *interface{}) (interface{}, error) {
 	bs, err := json.Marshal(x)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	err = UnmarshalJSON(bs, x)
+	if err != nil {
+		return nil, err
+	}
+	return *x, err
+}
+
+// JSONify recursively converts x into native-JSON types.
+// Deepcopies x, and does not mutate the underlying structure.
+func JSONify(x *interface{}) (interface{}, error) {
+	if x == nil {
+		return nil, nil
+	}
+
+	switch t := (*x).(type) {
+	case nil, bool, json.Number, int64, float64, string:
+		v := t
+		return v, nil
+
```

Comment:
I suppose we could add another faster path for `[]string` here (and other native-type slices), converting them to `[] interface {}`

Reply:
I'll try that on Tuesday. My benchmark uses string slices, and I could see it being a noticeable speedup.

Ditto map[string]x where x is a base type.

Reply:
Even without this, though, it doubles the throughput of adding objects to storage, so while it's worth doing it won't resolve the Gatekeeper issue on its own.

Reply:
Hrmm yeah perhaps we could use runtime reflection to turn the value-slices into slices of interface{}. But as you say, it's another improvement, if what we have here already improves the state of things, then that's good enough enough.